web/pdf_viewer.js

```diff
@@ -798,13 +802,15 @@ var PDFViewer = (function pdfViewer() {
      * @param {PageViewport} viewport
      * @returns {TextLayerBuilder}
      */
-    createTextLayerBuilder: function (textLayerDiv, pageIndex, viewport) {
+    createTextLayerBuilder: function (textLayerDiv, pageIndex, viewport,
+                                      enhanceTextSelection) {
       return new TextLayerBuilder({
         textLayerDiv: textLayerDiv,
         eventBus: this.eventBus,
         pageIndex: pageIndex,
         viewport: viewport,
-        findController: this.isInPresentationMode ? null : this.findController
+        findController: this.isInPresentationMode ? null : this.findController,
+        enhanceTextSelection: enhanceTextSelection
```

Comment:
cool! I updated the commit with the fixes you suggested. Also, I changed the logic around so that expand only gets called the first time you click on the text layer, and not on page render. I did this by turning bounds into a property of the object, and introducing the "expanded" flag to ensure expand only gets called once per page. As a result, I set the boolean to for enhanceTextSelection to true, since it only takes effect when a user highlights and otherwise has almost no effect. I can flip it back off if required.